e2e/cross_compilation/BUILD.bazel

```diff
@@ -0,0 +1,46 @@
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@toolchains_llvm_bootstrapped//constraints/libc:libc_versions.bzl", "LIBCS")
+load("@bazel_lib//lib:transitions.bzl", "platform_transition_binary")
+load("@bazel_skylib//rules:build_test.bzl", "build_test")
+
+cc_library(
+    name = "add",
+    srcs = ["add.c"],
+    hdrs = ["add.h"],
+    visibility = ["//visibility:public"],
+)
+
+cc_binary(
+    name = "main",
+    deps = [":add"],
+    srcs = ["test.cc"],
+)
+
+#TODO(cerisier): should the toolchain provide its own list ?
+PLATFORMS = [
+    "@toolchains_llvm_bootstrapped//platforms:linux_x86_64",
+    "@toolchains_llvm_bootstrapped//platforms:linux_aarch64",
+    # "@toolchains_llvm_bootstrapped//platforms:macos_x86_64",
+    # "@toolchains_llvm_bootstrapped//platforms:macos_aarch64",
+] + [
+    "@toolchains_llvm_bootstrapped//platforms/libc_aware:linux_x86_64_{}".format(libc) for libc in LIBCS
+] + [
+    "@toolchains_llvm_bootstrapped//platforms/libc_aware:linux_aarch64_{}".format(libc) for libc in LIBCS
+]
+
+[
+    platform_transition_binary(
```

Comment:
maybe we should have a macro here as well like `rust_binary_cross_build_test_suite`? Especially if we add more binaries like unwind.cc?

Reply:
I'll merge now as discussed and we we wil look at this when we introduce other tests.